cmd/zed/zed_event.c

```diff
@@ -628,6 +628,16 @@ _zed_event_add_nvpair(uint64_t eid, zed_strings_t *zsp, nvpair_t *nvp)
 	case DATA_TYPE_INT32:
 		(void) nvpair_value_int32(nvp, (int32_t *)&i32);
 		_zed_event_add_var(eid, zsp, prefix, name, "%d", i32);
+		/*
+		 * shadow readable strings for pool state
+		 */
+		if (strcmp(name, FM_EREPORT_PAYLOAD_ZFS_POOL_STATE) == 0) {
```

Comment:
How about changing the type to a `DATA_TYPE_UINT64` and handling it the same way as the other two existing exceptions FM_EREPORT_PAYLOAD_ZFS_VDEV_STATE and FM_EREPORT_PAYLOAD_ZFS_VDEV_LASTSTATE.